pkg/controller/machinepool/machinepool_controller.go

```diff
@@ -960,8 +960,9 @@ func (r *ReconcileMachinePool) updatePoolStatusForMachineSets(
 	}
 
 	// ...and taints
-	pool.Status.OwnedTaints = make([]hivev1.TaintIdentifier, len(pool.Spec.Taints))
-	for i, taint := range *controllerutils.GetUniqueTaints(&pool.Spec.Taints) {
+	uniqueTaints := *controllerutils.GetUniqueTaints(&pool.Spec.Taints)
+	pool.Status.OwnedTaints = make([]hivev1.TaintIdentifier, len(uniqueTaints))
```

Comment:
What was the effect before this fix? Would OwnedTaints end up with trailing empty entries at the end?

```yaml
  ownedTaints:
  - key: foo
    effect: bar
  - key: ""
    effect: ""
```
?

Can we add unit test coverage for this case? I guess by adding a length check, and/or verifying all taintEffects are non-"empty"?

Reply:
Okay so you were totally correct, there were empty entries just like you mentioned. I added some unit tests to ensure we don't run into it again - I took the easy way out and split the setting of OwnedLabels and OwnedTaints in a separate function. I'm also explicit about length check to not run into nil vs empty comparisons.